server.lua

```diff
@@ -1,9 +1,19 @@
+local function uuid()
+	local random = math.random
+    local template ='xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'
+    return string.gsub(template, '[xy]', function (c)
+        local v = (c == 'x') and random(0, 0xf) or random(8, 0xb)
+        return string.format('%x', v)
+    end)
+end
+
 function playSound(toBePlayedOn, audioSource, audioCoord, audioRotation)
-	TriggerClientEvent('3dsound:playSound', toBePlayedOn, audioSource, audioCoord, audioRotation)	
+	local uid = uuid()
+	TriggerClientEvent('3dsound:playSound', toBePlayedOn, audioSource, audioCoord, audioRotation, uid)	
```

Comment:
We can store all the generated `uid`'s in a table which can be used to check if the audio-id exists when `stopSound` export is called. It also helps maintaining the server-side validation for authentic id's.

Reply:
Okay, sounds good